x/zoneconcierge/keeper/channel_keeper.go

```diff
@@ -0,0 +1,146 @@
+package keeper
+
+import (
+	"context"
+
+	"cosmossdk.io/collections"
+	corestoretypes "cosmossdk.io/core/store"
+	"cosmossdk.io/log"
+	"github.com/babylonlabs-io/babylon/v3/x/zoneconcierge/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	channeltypes "github.com/cosmos/ibc-go/v10/modules/core/04-channel/types"
+	ibcexported "github.com/cosmos/ibc-go/v10/modules/core/exported"
+)
+
+type (
+	ChannelKeeper struct {
+		cdc           codec.BinaryCodec
+		storeService  corestoretypes.KVStoreService
+		channelKeeper types.ChannelKeeper
+
+		// Collections for KV store management
+		Schema collections.Schema
+		port   collections.Item[string]
+	}
+)
+
+func NewChannelKeeper(
+	cdc codec.BinaryCodec,
+	storeService corestoretypes.KVStoreService,
+	channelKeeper types.ChannelKeeper,
+) *ChannelKeeper {
+	sb := collections.NewSchemaBuilder(storeService)
+
+	k := &ChannelKeeper{
+		cdc:           cdc,
+		storeService:  storeService,
+		channelKeeper: channelKeeper,
+
+		port: collections.NewItem[string](
+			sb,
+			types.PortKey,
+			"port",
+			collections.StringValue,
+		),
+	}
+
+	schema, err := sb.Build()
+	if err != nil {
+		panic(err)
+	}
+	k.Schema = schema
+
+	return k
+}
+
+// Logger returns a module-specific logger.
+func (k ChannelKeeper) Logger(ctx sdk.Context) log.Logger {
+	return ctx.Logger().With("module", "x/"+ibcexported.ModuleName+"-"+types.ModuleName+"-channel")
+}
+
+// InitGenesis initializes the keeper state from a provided initial genesis state.
+func (k ChannelKeeper) InitGenesis(ctx context.Context, gs types.GenesisState) error {
```

Comment:
it got dangerous to combine this keeper inside another keeper, but this might be fix/changed in the future